commands/job/options.go

```diff
@@ -0,0 +1,147 @@
+package job
+
+import (
+	"fmt"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/lily/chain/indexer/tasktype"
+	"github.com/filecoin-project/lily/lens/lily"
+)
+
+var log = logging.Logger("lily/commands/job")
+
+type runOpts struct {
+	Storage string
+	Name    string
+
+	Tasks cli.StringSlice
+
+	Window       time.Duration
+	RestartDelay time.Duration
+
+	RestartCompletion bool
+	RestartFailure    bool
+}
+
+func (r runOpts) ParseJobConfig(kind string) lily.LilyJobConfig {
+	if RunFlags.Name == "" {
+		RunFlags.Name = fmt.Sprintf("%s_%d", kind, time.Now().Unix())
+	}
+	return lily.LilyJobConfig{
+		Name:                RunFlags.Name,
+		Storage:             RunFlags.Storage,
+		Tasks:               RunFlags.Tasks.Value(),
+		Window:              RunFlags.Window,
+		RestartOnFailure:    RunFlags.RestartFailure,
+		RestartOnCompletion: RunFlags.RestartCompletion,
+		RestartDelay:        RunFlags.RestartDelay,
+	}
+}
+
+var RunFlags runOpts
+
+var RunWindowFlag = &cli.DurationFlag{
+	Name:        "window",
+	Usage:       "Duaration after which job execution will be canceled",
+	EnvVars:     []string{"LILY_JOB_WINDOW"},
+	Value:       0,
+	Destination: &RunFlags.Window,
+}
+
+var RunTaskFlag = &cli.StringSliceFlag{
+	Name:        "tasks",
+	Usage:       "Comma separated list of tasks to run in job. Each task is reported separately in the storage backend.",
+	EnvVars:     []string{"LILY_JOB_TASKS"},
+	Value:       cli.NewStringSlice(tasktype.AllTableTasks...),
```

Comment:
by default we enable all tasks.
This means calls to the survey command will error when passed the default values, and callers will need to explicit set tasks to its expected values - `peeragents`

Reply:
That's not a huge deal. But ideally there's an override Flag which inherits the regular taskFlag fields but sets the Value appropriately. I leave this up to you.

Reply:
Users can still do that with this flag: `lily job run --tasks=peeragents survey` overrides the default values of this flag, as does `lily job run --task=block_header watch`.